python/ray/tests/test_resource_demand_scheduler.py

```diff
@@ -365,6 +367,108 @@ def test_calculate_node_resources():
     assert to_launch == {"p2.8xlarge": 1}
 
 
+def test_backlog_queue_impact_on_binpacking_time():
+    new_types = copy.deepcopy(TYPES_A)
+    new_types["p2.8xlarge"]["max_workers"] = 1000
+    new_types["m4.16xlarge"]["max_workers"] = 1000
+
+    def test_backlog_queue_impact_on_binpacking_time_aux(
+            num_available_nodes, time_to_assert, demand_request_shape):
+        provider = MockProvider()
+        scheduler = ResourceDemandScheduler(
+            provider, new_types, max_workers=10000)
+
+        provider.create_node({}, {
+            TAG_RAY_USER_NODE_TYPE: "m4.16xlarge",
+            TAG_RAY_NODE_STATUS: STATUS_UP_TO_DATE
+        }, num_available_nodes)
+        # <num_available_nodes> m4.16xlarge instances.
+        cpu_ips = provider.non_terminated_node_ips({})
+        provider.create_node({}, {
+            TAG_RAY_USER_NODE_TYPE: "p2.8xlarge",
+            TAG_RAY_NODE_STATUS: STATUS_UP_TO_DATE
+        }, num_available_nodes)
+        # <num_available_nodes>  m4.16xlarge and <num_available_nodes>
+        # p2.8xlarge instances.
+        all_nodes = provider.non_terminated_nodes({})
+        all_ips = provider.non_terminated_node_ips({})
+        gpu_ips = [ip for ip in all_ips if ip not in cpu_ips]
+        usage_by_ip = {}
+        # 2x<num_available_nodes> free nodes (<num_available_nodes> m4.16xlarge
+        # and <num_available_nodes> p2.8xlarge instances).
+        for i in range(num_available_nodes):
+            usage_by_ip[cpu_ips[i]] = {"CPU": 64}
+            usage_by_ip[gpu_ips[i]] = {"GPU": 8, "CPU": 32}
+        demands = demand_request_shape * \
+            AUTOSCALER_MAX_RESOURCE_DEMAND_VECTOR_SIZE
+        t1 = time.time()
+        to_launch = scheduler.get_nodes_to_launch(all_nodes, {}, demands,
+                                                  usage_by_ip, [])
+        t2 = time.time()
+        assert t2 - t1 < time_to_assert
+        return to_launch
+
+    # The times asserted below are 2x the actual time took when this test
+    # was measured on 2.3 GHz 8-Core Intel (I9-9880H) Core i9.
+
+    # Check the time it takes when there are 0 nodes available and the demand
+    # is requires adding another ~100 nodes.
+    to_launch = test_backlog_queue_impact_on_binpacking_time_aux(
+        num_available_nodes=0,
+        time_to_assert=0.4,
```

Comment:
Small timeouts are likely to cause flaky tests if the machine happens to be busy--- how about we raise to 10s as a sanity check, and just document how long it should take?